src/common/providers/identity.ts

```diff
@@ -640,9 +647,38 @@ function parseAdditionalUserInfo(decodedJWT: DecodedPayload): AdditionalUserInfo
     profile,
     username,
     isNewUser: decodedJWT.event_type === "beforeCreate" ? true : false,
+    recaptchaScore: decodedJWT.recaptcha_score,
   };
 }
 
+/** Helper to generate payload to GCIP from client request.
+ * @internal
+ */
+export function generateRequestPayload(
+  authResponse?: BeforeCreateResponse | BeforeSignInResponse
+): any {
```

Comment:
Do we know exactly the shape of the data GCIP expects? If we do then we could statically type the return value here instead of using `any`, otherwise not really a huge deal.

Reply:
+1, let's avoid 'any' if possible. I guess BeforeSignInResponse is the type to use? (Or if we consolidate all fields into BlockingFunctionsResponse we could use that, in future)

Reply:
BeforeSignInResponse is the interface for blocking function returns for developers (e.g. developers can return displayName, emailVeried etc field when they implemented the blocking function).  The requestPayload(responsePayload) here is (will be) a new different interface that follows GCIP backend proto.  They are very similar but nested a bit differently.  This is also the reason why the naming is kind of confusing, it is generated from the blocking function response as a request to GCIP backend.

Reply:
Added a new interface for this.